src/realm/object-store/impl/deep_change_checker.hpp

```diff
@@ -50,7 +50,7 @@ struct CollectionChangeInfo {
 struct TransactionChangeInfo {
     std::vector<CollectionChangeInfo> collections;
     std::unordered_map<TableKey, ObjectChangeSet> tables;
-    bool schema_changed;
+    bool schema_changed = false;
```

Comment:
This should be OK, we always check whether this flag is `true` or not, but it is nice to have it init in the struct.

Reply:
This is the stacktrace from the test realm-object-store-tests "[dictionary]" -c "notifications" -c "now with links":
```
    frame #5: 0x000000010171f1a4 realm-object-store-tests`realm::_impl::CollectionNotifier::get_modification_checker(this=0x0000000107908808, info=0x000000016fdfa260, root_table=ConstTableRef @ 0x000000016fdf9ee8) at collection_notifier.cpp:51:14 [opt]
    frame #6: 0x00000001017409a4 realm-object-store-tests`realm::_impl::ListNotifier::run(this=0x0000000107908808) at list_notifier.cpp:99:34 [opt]
    frame #7: 0x000000010174f918 realm-object-store-tests`realm::_impl::RealmCoordinator::run_async_notifiers(this=0x0000000107905388) at realm_coordinator.cpp:981:19 [opt]
    frame #8: 0x000000010174e844 realm-object-store-tests`realm::_impl::RealmCoordinator::on_change(this=0x0000000107905388) at realm_coordinator.cpp:839:9 [opt]
    frame #9: 0x000000010101b9b8 realm-object-store-tests`on_change_but_no_notify(realm=0x0000000107905858) at test_file.cpp:498:68 [opt]
    frame #10: 0x000000010101ba14 realm-object-store-tests`advance_and_notify(realm=0x0000000107905858) at test_file.cpp:503:5 [opt]
    frame #11: 0x000000010008bbe8 realm-object-store-tests`void CATCH2_INTERNAL_TEMPLATE_TEST_0<realm::collection_fixtures::MixedVal>() at dictionary.cpp:774:13 [opt]
...
(lldb) fr s 11
warning: realm-object-store-tests was compiled with optimization - stepping may behave oddly; variables may not be available.
frame #11: 0x000000010008bbe8 realm-object-store-tests`void CATCH2_INTERNAL_TEMPLATE_TEST_0<realm::collection_fixtures::MixedVal>() at dictionary.cpp:774:13 [opt]
   771              links.insert("l", another.get_key());
   772              links.insert("m", ObjKey());
   773              r->commit_transaction();
-> 774              advance_and_notify(*r);
   775              REQUIRE(local_change.insertions.count() == 2);
   776
   777              SECTION("with links on frozen Realm") {
(lldb) do
frame #10: 0x000000010101ba14 realm-object-store-tests`advance_and_notify(realm=0x0000000107905858) at test_file.cpp:503:5 [opt]
   500
   501  void advance_and_notify(Realm& realm)
   502  {
-> 503      on_change_but_no_notify(realm);
   504      realm.notify();
   505  }
   506  #endif
(lldb)
frame #9: 0x000000010101b9b8 realm-object-store-tests`on_change_but_no_notify(realm=0x0000000107905858) at test_file.cpp:498:68 [opt]
   495
   496  void on_change_but_no_notify(Realm& realm)
   497  {
-> 498      _impl::RealmCoordinator::get_coordinator(realm.config().path)->on_change();
   499  }
   500
   501  void advance_and_notify(Realm& realm)
(lldb)
frame #8: 0x000000010174e844 realm-object-store-tests`realm::_impl::RealmCoordinator::on_change(this=0x0000000107905388) at realm_coordinator.cpp:839:9 [opt]
   836
   837      {
   838          util::CheckedUniqueLock lock(m_running_notifiers_mutex);
-> 839          run_async_notifiers();
   840      }
   841
   842      util::CheckedLockGuard lock(m_realm_mutex);
(lldb)
frame #7: 0x000000010174f918 realm-object-store-tests`realm::_impl::RealmCoordinator::run_async_notifiers(this=0x0000000107905388) at realm_coordinator.cpp:981:19 [opt]
   978      // Change info is now all ready, so the notifiers can now perform their
   979      // background work
   980      for (auto& notifier : notifiers) {
-> 981          notifier->run();
   982      }
   983
   984      // Reacquire the lock while updating the fields that are actually read on
(lldb)
frame #6: 0x00000001017409a4 realm-object-store-tests`realm::_impl::ListNotifier::run(this=0x0000000107908808) at list_notifier.cpp:99:34 [opt]
   96       m_prev_size = m_list->size();
   97
   98       if (m_info && m_type == PropertyType::Object) {
-> 99           auto object_did_change = get_modification_checker(*m_info, m_list->get_target_table());
   100          for (size_t i = 0; i < m_prev_size; ++i) {
   101              if (m_change.modifications.contains(i))
   102                  continue;
(lldb)
frame #5: 0x000000010171f1a4 realm-object-store-tests`realm::_impl::CollectionNotifier::get_modification_checker(this=0x0000000107908808, info=0x000000016fdfa260, root_table=ConstTableRef @ 0x000000016fdf9ee8) at collection_notifier.cpp:51:14 [opt]
   48       // related tables info. This'll also happen for schema changes that don't
   49       // matter, but making this check more precise than "any schema change at all
   50       // happened" would mostly just be a source of potential bugs.
-> 51       if (info.schema_changed) {
   52           util::CheckedLockGuard lock(m_callback_mutex);
   53           update_related_tables(*root_table);
   54       }
```